src/Analyser/ArgumentsNormalizer.php

```diff
@@ -76,6 +76,64 @@ public static function reorderCallUserFuncArguments(
 		)];
 	}
 
+	/**
+	 * @return array{ParametersAcceptor, FuncCall}|null
+	 */
+	public static function reorderCallUserFuncArrayArguments(
+		FuncCall $callUserFuncCall,
+		Scope $scope,
+	): ?array
+	{
+		$args = $callUserFuncCall->getArgs();
+		if (count($args) < 1 || count($args) > 2) {
+			return null;
+		}
+
+		$callbackArg = null;
+		$passThruArgs = [];
+		foreach ($args as $i => $arg) {
+			if ($callbackArg === null) {
+				if ($arg->name === null && $i === 0) {
+					$callbackArg = $arg;
+					continue;
+				}
+				if ($arg->name !== null && $arg->name->toString() === 'callback') {
+					$callbackArg = $arg;
+					continue;
+				}
+			}
+
+			$passThruArgs[] = new Arg(
+				$arg->value,
+				$arg->byRef,
+				true,
```

Comment:
Does this work solely thanks to `unpack` set to `true`? I don't really like this, we should process this ourselves. One reason is that if `$arg->unpack` in the original `call_user_func_array` is already true, this will lead to wrong results.

The easiest example is when the arg is just a single constant array, right. But we can also have multiple constant arrays in union, and also general arrays.

Reply:
so to get it right, in case of

```php
	/**
	 * @param array{i: 5}|array{i: 15} $arr
	 */
	public function doFoo(array $arr): void
	{
		$f = function (int $i): void {

		};
		call_user_func_array(callback: $f, args: $arr);
	}
```

the argument normalizer needs to return multiple FuncCalls?

Reply:
No. We just need to evaluate as if the callback was called like this:

```php
$f($i);
```

Where `$i` is `5|15`.